gulpfile.js

```diff
@@ -60,3 +61,29 @@ gulp.task(
     })
   )
 );
+
+gulp.task('update-patterns-index', async () => {
+  // Determine the name of the JS/CSS bundles (in prod they could contain a hash)
+  const jsBundleName = fs
+    .readdirSync('build/scripts')
+    .find(file => file.startsWith('pattern-library'));
+  const cssBundleName = fs
+    .readdirSync('build/styles')
+    .find(file => file.startsWith('pattern-library'));
+
+  if (!jsBundleName || !cssBundleName) {
+    throw new Error(
+      'JS and/or CSS bundles missing. Make sure bundles are generated first'
+    );
+  }
+
+  // Replace references to the generic bundle names with the resolved ones
+  const indexContent = fs
+    .readFileSync('templates/index.html')
+    .toString()
+    .replace('pattern-library.bundle.js', jsBundleName)
+    .replace('pattern-library.css', cssBundleName);
+
+  // eslint-disable-next-line no-undef
+  fs.writeFileSync('templates/index.html', Buffer.from(indexContent));
```

Comment:
Here `templates/index.html` has become an actual template instead of just a static file. If that is the case then I think we should make that obvious by rendering it with a template renderer when served or at build time. Assuming we're continuing to serve this as static content, then the output should go into the `build/` directory rather than modifying the source file.

Reply:
The dumbest possible template renderer is to use some obvious placeholder like `{{ js_bundle }}` and do a string search and replace all. We might as well use something like Mustache though and at least get proper escaping.

Reply:
> Here templates/index.html has become an actual template instead of just a static file. If that is the case then I think we should make that obvious by rendering it with a template renderer when served or at build time. Assuming we're continuing to serve this as static content, then the output should go into the build/ directory rather than modifying the source file.

We recently renamed this file to change the mustache-specific extension to `.html` because it was static. If it's a "template again," we should once again change the source file name, methinks.